Lib/test/test_py_compile.py

```diff
@@ -49,6 +49,9 @@ def __new__(mcls, name, bases, dct, *, source_date_epoch):
 
         return cls
 
+    def __init__(cls, name, bases, dct, *, source_date_epoch):
+        super().__init__(name, bases, dct)
+
```

Comment:
add `__init__` to prevent `source_date_epoch` from being passed through to `__init_subclass__`